controlplane/rosa/controllers/rosacontrolplane_controller.go

```diff
@@ -352,6 +365,115 @@ func (r *ROSAControlPlaneReconciler) reconcileDelete(ctx context.Context, rosaSc
 	return ctrl.Result{}, nil
 }
 
+func (r *ROSAControlPlaneReconciler) reconcileKubeconfig(ctx context.Context, rosaScope *scope.ROSAControlPlaneScope, rosaClient *rosa.RosaClient, cluster *cmv1.Cluster) (ctrl.Result, error) {
```

Comment:
Add comments inline to explain the business logic?

Reply:
done